documentation/modules/ROOT/pages/development/engine.adoc

```diff
@@ -480,6 +513,44 @@ The default is a periodic commity policy based upon time intervals.
 |Max delay (in ms) between retries when encountering connection errors.
 |===
 
+[[async-engine-properties]]
+=== Asynchronous Engine Properties
+
+[cols="35%a,10%a,55%a"]
+|===
+|Property
+|Default
+|Description
+
+|`record.processing.threads`
+|`#` of availabel cores
+|The number of threads to be used for processing CDC records. The default is number of available machine cores with upper limit of 16 threads.
+If you want to use all available thread without any limitation, use `AVAILABLE_CORES` placeholder.
+
+|`record.processing.shutdown.timeout.ms`
+|1000
+|Maximum time in milliseconds to wait for processing submitted records when task shutdown is called.
+
+|`record.processing.order`
+|`ORDERED`
+|Determines how the records should be produced.
+
+* `ORDERED` means sequential processing, i.e. that the records are produced in the same order as they were obtained from the database.
+* `UNORDERED` means non-sequential processing, i.e. the records can be produced in a different order than the original one.
+
+Non-sequential approach gives better throughput, as the records are produced immediately once the SMTs and serialization of the message is done, without waiting of other records.
+This option doesn't have any effect when ChangeConsumer is provided to the engine.
+
+|`record.processing.with.serial.consumer`
+|`false`
+|Specifies whether the default ChangeConsumer should be created from provided Consumer, resulting in serial Consumer processing.
+This option has no effect if the ChangeConsumer is already provided to the engine via configuration.
+
+|`task.management.timeout.ms`
+|180,000 (3 min)
+|Time to wait for task's lifecycle management operations (starting and stopping), given in milliseconds.
```

Comment:
```suggestion
|Time, in milliseconds, that the engine waits for a task's lifecycle management operations (starting and stopping) to complete.
```
What happens if the specified time is exceeded?

Reply:
Engine throws an exception, which results in stopping the engine, so if the time is exceed during start, engine fails to start, in case of shutdown, engine would be stopped abruptly (e.g. before processing all buffered records, not storing latest offset etc.)